api/src/wfl/module/covid.clj

```diff
@@ -94,82 +96,122 @@
+                     (source/stop-source! tx source)
+                     (patch-workload tx workload {:stopped now :updated now})
+                     (when-not (:started workload)
+                               (patch-workload tx workload {:finished now}))
+                     (workloads/load-workload-for-id tx id))]
+             (when-not started
+                       (throw (UserException. "Cannot stop workload before it's been started."
+                                              {:workload workload})))
+             (if-not (or stopped finished) (stop! workload (utc-now)) workload)))
 
 (defn ^:private workload-to-edn [workload]
-  (-> workload
-      (util/select-non-nil-keys workload-metadata-keys)
-      (dissoc :pipeline)
-      (update :source   util/to-edn)
-      (update :executor util/to-edn)
-      (update :sink     util/to-edn)))
-
-(defoverload workloads/create-workload!    pipeline create-covid-workload)
-(defoverload workloads/start-workload!     pipeline start-covid-workload)
-(defoverload workloads/update-workload!    pipeline update-covid-workload)
-(defoverload workloads/stop-workload!      pipeline stop-covid-workload)
-(defoverload workloads/retry               pipeline batch/retry-unsupported)
-(defoverload workloads/load-workload-impl  pipeline load-covid-workload-impl)
-(defmethod   workloads/workflows           pipeline
-  [tx {:keys [executor] :as _workload}]
-  (executor/executor-workflows tx executor))
-(defmethod   workloads/workflows-by-status pipeline
-  [tx {:keys [executor] :as _workload} status]
-  (executor/executor-workflows-by-status tx executor status))
-(defoverload workloads/to-edn             pipeline workload-to-edn)
+      (-> workload
+          (util/select-non-nil-keys workload-metadata-keys)
+          (dissoc :pipeline)
+          (update :source util/to-edn)
+          (update :executor util/to-edn)
+          (update :sink util/to-edn)))
+
+(defn retry-workflows
+      [workload workflows]
+      ; Get the workflows from the terra executor details table
+      ;(->> workload :executor :details
+      ;     (format "SELECT * FROM %s")
+      ;     (jdbc/query (postgres/wfl-db-config))
+      ;     (wfl.debug/trace)
+      ;     )
+
+      ; Get the rows from the terra details table for the workflows
+      ; iterate through the results and get the distinct reference values
+      ; resubmit the reference values
+      ; write back to the table, putting the new workflow id into the retry field of the failed one
+      (let [workflow_ids (util/to-quoted-comma-separated-list (map :uuid workflows))
+            executor (get-in workload [:executor])
+            executor_details (get-in workload [:executor :details])
+            query (format "SELECT * FROM %s WHERE workflow IN (%s)" executor_details workflow_ids)
+            results (jdbc/query (postgres/wfl-db-config) [query])
+            distinct_references (set (map :reference results))]
```

Comment:
This gives you the distinct reference IDs but you want the snapshot reference objects from Rawls to pass to `executor/create-submission!` (more context below). 

Suggest:
```
distinct_references (->> (map :reference results)
                                 set
                                 (map (partial rawls/get-snapshot-reference workspace)))
```